src/components/Tokens/TokenDetails/TokenDetail.tsx

```diff
@@ -161,27 +268,95 @@ export default function LoadedTokenDetail({ address }: { address: string }) {
   const chainInfo = getChainInfo(token?.chainId)
   const networkLabel = chainInfo?.label
   const networkBadgebackgroundColor = chainInfo?.backgroundColor
+  const environment = useRelayEnvironment()
+  const queryReference = loadQuery(environment, tokenDetailsStatsQuery, {
+    contract: {
+      address,
+      chain: token ? token.chainId : SupportedChainId.MAINNET,
```

Comment:
i'm not sure falling back to mainnet is right here - imo the address + chainId go together as a unit, and in the worst case we would show the wrong token on a different network